source/renderer/app/features/discreet-mode/api/index.js

```diff
@@ -1,14 +1,20 @@
 // @flow
-import LocalStorageApi from '../../../api/utils/localStorage';
+import { localStorageBridge } from '../../../api/utils/localStorageBridge';
 import { STORAGE_KEYS as storageKeys } from '../../../../../common/config/electron-store.config';
 
 export class DiscreetModeApi {
-  getDiscreetModeSettings = (): Promise<boolean> =>
-    LocalStorageApi.get(storageKeys.DISCREET_MODE_ENABLED, false);
+  getDiscreetModeSettings = async (): Promise<boolean> => {
+    const localStorage = await localStorageBridge();
```

Comment:
It would probably be better to pass the localstorage API as a constructor argument to `DiscreetModeApi` otherwise this is hard wired

Reply:
Do we really need to use dependency injection in this case? The feature is pretty small and self-contained and not leaking to other domains.

Also, what would you pass to the constructor? A the bridge ::`Promise<{get, set, etc...}>` or the instance itself? 

If the instance is used then `const [discreetModeFeature] = useState<DiscreetMode>` would have to be refactored to use `useEffect` because it has to resolve the promise first. 

I would prefer to keep like this but i'm open to suggestions of how to inject this value.

Reply:
Yeah, I think declaring the simplest interface for the required API methods is the way to go here. Regarding the eager vs lazy loading: since the electron local storage solution is already eagerly loaded + static we could replace the current `localStorageBridge` function with its contents (simple `if (isElectron)` then local storage is the electron storage, web  otherwise)